WIP Python/PHP CentOS install notes

RichardSmedley opened the pull request:
Something to be surfaced in install instructions and in compatibility guides

DOC-10111 /   DOC-10115

RichardSmedley committed:
WIP Python/PHP CentOS install notes

Something to be surfaced in install instructions and in compatibility guides

DOC-10111 /   DOC-10115

maria-robobug commented on modules/pages/partials/docker.adoc:
```diff
@@ -0,0 +1,8 @@
+This SDK runs on top of the {cpp} core, Couchbase++, which requires a {cpp} 17 compiler, such as GCC 8.0, or more recent versions.
```
We can use the {[cbpp](https://github.com/couchbase/docs-site/blob/master/antora-playbook.yml#L161)} variable for `Couchbase++`

RichardSmedley committed:
{cbpp}

maria-robobug commented on modules/pages/partials/docker.adoc:
```diff
@@ -0,0 +1,8 @@
+This SDK runs on top of the {cpp} core, {cbpp}, which requires a {cpp} 17 compiler, such as GCC 8.0, or more recent versions.
+Centos 7 ships with an older version of GCC.
+
+Workarounds are available for installing a newer build chain with https://docs.fedoraproject.org/en-US/epel/[EPEL], 
+but this may not be within your company's policy, so also take a look at container options -- 
+such as the unofficial Docker builds https://github.com/raycardillo/couchbase-dockerfiles[here], 
+which can at least be used as a reference to a known working set-up, 
+to attempt a version specific to your circumstances, and any subsequent change in dependencies.
```
Maybe this should be presented as **WARNING** / **IMPORTANT** , since we are suggesting the usage of unofficial Docker images.

LGTM overall, not sure if we need the last bit though: `to attempt a version specific to your circumstances, and any subsequent change in dependencies`. 
Seems implicit that users will change the reference Dockerfiles to suit their needs.

```suggestion
WARNING: Workarounds are available for installing a newer build chain with https://docs.fedoraproject.org/en-US/epel/[EPEL], 
but this may not be within your company's policy, so also take a look at container options -- 
such as the unofficial Docker builds https://github.com/raycardillo/couchbase-dockerfiles[here], 
which can at least be used as a reference to a known working set-up.
```

osfameron replied:
Hmm, I've added a few dockerfiles to sdk repos, without flagging them as "official" (but I wonder if their mere presence in the repo implies it?)

Might be worth seeing if @raycardillo wants to fork that repo to couchbaselabs? It could still be marked unofficial, but best-endeavours, and at least it's *supplied* by Couchbase, so known to be in-good-faith as a good starting point?

RichardSmedley replied:
It's a possibility.
I was starting from trying not to add to anyone's tasks, but if Ray has the bandwidth to do this, then we can update the link accordingly.  :+1:

RichardSmedley committed:
Admonition

Co-authored-by: Maria Shodunke <maria-robobug@users.noreply.github.com>

RichardSmedley committed:
warning  -->  important

RichardSmedley committed:
Gah, typo!

Keyboardologists will be able to detect which non-Qwerty keyboard the writer uses.  ;)
